src/core/variable.jl

```diff
@@ -299,6 +299,60 @@ function variable_reactive_dcline_flow(pm::GenericPowerModel; bounded = true)
     end
 end
 
+"variable: `va_shift[l,i,j]` for `(l,i,j)` in `arcs`"
+function variable_phase_shift(pm::GenericPowerModel; bounded = true)
+
+    shift_min = Dict()
+    shift_max = Dict()
+    shift = Dict()
+    for (l,i,j) in pm.ref[:arcs_from]
+        shift_min[(l,i,j)] = pm.ref[:branch][l]["shift_fr_min"]
+        shift_max[(l,i,j)] = pm.ref[:branch][l]["shift_fr_max"]
+        shift[(l,i,j)] = pm.ref[:branch][l]["shift_fr"]
+    end
+    for (l,i,j) in pm.ref[:arcs_to]
+        shift_min[(l,i,j)] = pm.ref[:branch][l]["shift_to_min"]
+        shift_max[(l,i,j)] = pm.ref[:branch][l]["shift_to_max"]
+        shift[(l,i,j)] = pm.ref[:branch][l]["shift_to"]
+    end

```

Comment:
If these computations are used some other place, you may consider adding them to` pm.ext`